packages/interflex/src/locale.ts

```diff
@@ -0,0 +1,581 @@
+  | 'lv_LV'
+  | 'ln'
+  | 'ln_AO'
+  | 'ln_CF'
+  | 'ln_CG'
+  | 'ln_CD'
+  | 'lt'
+  | 'lt_LT'
+  | 'lu'
+  | 'lu_CD'
+  | 'lb'
+  | 'lb_LU'
+  | 'mk'
+  | 'mk_MK'
+  | 'mg'
+  | 'mg_MG'
+  | 'ms'
+  | 'ms_BN'
+  | 'ms_Latn_BN'
+  | 'ms_Latn_MY'
+  | 'ms_Latn_SG'
+  | 'ms_Latn'
+  | 'ms_MY'
+  | 'ms_SG'
+  | 'ml'
+  | 'ml_IN'
+  | 'mt'
+  | 'mt_MT'
+  | 'gv'
+  | 'gv_IM'
+  | 'mr'
+  | 'mr_IN'
+  | 'mn'
+  | 'mn_Cyrl_MN'
+  | 'mn_Cyrl'
+  | 'mn_MN'
+  | 'ne'
+  | 'ne_IN'
+  | 'ne_NP'
+  | 'nd'
+  | 'nd_ZW'
+  | 'se'
+  | 'se_FI'
+  | 'se_NO'
+  | 'se_SE'
+  | 'no'
+  | 'no_NO'
+  | 'nb'
+  | 'nb_NO'
+  | 'nb_SJ'
+  | 'nn'
+  | 'nn_NO'
+  | 'or'
+  | 'or_IN'
+  | 'om'
+  | 'om_ET'
+  | 'om_KE'
+  | 'os'
+  | 'os_GE'
+  | 'os_RU'
+  | 'ps'
+  | 'ps_AF'
+  | 'fa'
+  | 'fa_AF'
+  | 'fa_IR'
+  | 'pl'
+  | 'pl_PL'
+  | 'pt'
+  | 'pt_AO'
+  | 'pt_BR'
+  | 'pt_CV'
+  | 'pt_GW'
+  | 'pt_MO'
+  | 'pt_MZ'
+  | 'pt_PT'
+  | 'pt_ST'
+  | 'pt_TL'
+  | 'pa'
+  | 'pa_Arab_PK'
+  | 'pa_Arab'
+  | 'pa_Guru_IN'
+  | 'pa_Guru'
+  | 'pa_IN'
+  | 'pa_PK'
+  | 'qu'
+  | 'qu_BO'
+  | 'qu_EC'
+  | 'qu_PE'
+  | 'ro'
+  | 'ro_MD'
+  | 'ro_RO'
+  | 'rm'
+  | 'rm_CH'
+  | 'rn'
+  | 'rn_BI'
+  | 'ru'
+  | 'ru_BY'
+  | 'ru_KZ'
+  | 'ru_KG'
+  | 'ru_MD'
+  | 'ru_RU'
+  | 'ru_UA'
+  | 'sg'
+  | 'sg_CF'
+  | 'gd'
+  | 'gd_GB'
+  | 'sr'
+  | 'sr_BA'
+  | 'sr_Cyrl_BA'
+  | 'sr_Cyrl_XK'
+  | 'sr_Cyrl_ME'
+  | 'sr_Cyrl_RS'
+  | 'sr_Cyrl'
+  | 'sr_XK'
+  | 'sr_Latn_BA'
+  | 'sr_Latn_XK'
+  | 'sr_Latn_ME'
+  | 'sr_Latn_RS'
+  | 'sr_Latn'
+  | 'sr_ME'
+  | 'sr_RS'
+  | 'sh'
+  | 'sh_BA'
+  | 'sn'
+  | 'sn_ZW'
+  | 'ii'
+  | 'ii_CN'
+  | 'si'
+  | 'si_LK'
+  | 'sk'
+  | 'sk_SK'
+  | 'sl'
+  | 'sl_SI'
+  | 'so'
+  | 'so_DJ'
+  | 'so_ET'
+  | 'so_KE'
+  | 'so_SO'
+  | 'es'
+  | 'es_AR'
+  | 'es_BO'
+  | 'es_IC'
+  | 'es_EA'
+  | 'es_CL'
+  | 'es_CO'
+  | 'es_CR'
+  | 'es_CU'
+  | 'es_DO'
+  | 'es_EC'
+  | 'es_SV'
+  | 'es_GQ'
+  | 'es_GT'
+  | 'es_HN'
+  | 'es_MX'
+  | 'es_NI'
+  | 'es_PA'
+  | 'es_PY'
+  | 'es_PE'
+  | 'es_PH'
+  | 'es_PR'
+  | 'es_ES'
+  | 'es_US'
+  | 'es_UY'
+  | 'es_VE'
+  | 'sw'
+  | 'sw_KE'
+  | 'sw_TZ'
+  | 'sw_UG'
+  | 'sv'
+  | 'sv_AX'
+  | 'sv_FI'
+  | 'sv_SE'
+  | 'tl'
+  | 'tl_PH'
+  | 'ta'
+  | 'ta_IN'
+  | 'ta_MY'
+  | 'ta_SG'
+  | 'ta_LK'
+  | 'te'
+  | 'te_IN'
+  | 'th'
+  | 'th_TH'
+  | 'bo'
+  | 'bo_CN'
+  | 'bo_IN'
+  | 'ti'
+  | 'ti_ER'
+  | 'ti_ET'
+  | 'to'
+  | 'to_TO'
+  | 'tr'
+  | 'tr_CY'
+  | 'tr_TR'
+  | 'uk'
+  | 'uk_UA'
+  | 'ur'
+  | 'ur_IN'
+  | 'ur_PK'
+  | 'ug'
+  | 'ug_Arab_CN'
+  | 'ug_Arab'
+  | 'ug_CN'
+  | 'uz'
+  | 'uz_AF'
+  | 'uz_Arab_AF'
+  | 'uz_Arab'
+  | 'uz_Cyrl_UZ'
+  | 'uz_Cyrl'
+  | 'uz_Latn_UZ'
+  | 'uz_Latn'
+  | 'uz_UZ'
+  | 'vi'
+  | 'vi_VN'
+  | 'cy'
+  | 'cy_GB'
+  | 'fy'
+  | 'fy_NL'
+  | 'yi'
+  | 'yo'
+  | 'yo_BJ'
+  | 'yo_NG'
+  | 'zu'
+  | 'zu_ZA';
+
+type LocaleData = {
+  localizedName: string;
+  currencySymbol: string;
+  thousandSeparator: string;
+  decimalSeparator: string;
+  decimalPlaces: number;
```

Comment:
`thousandSeparator` and `decimalSeparator` are related to formatNumber which is not included in the PR, so maybe we should leave those out.
`decimalPlaces` is related to currency, if I recall correctly, so that goes with `currencySymbol` in a future PR with `formatCurrency()` (and importantly, I want to be careful not to tightly couple currency with locale because locale is more like a display choice that the user makes, but currency is often NOT a user-choice, e.g. I can't choose what currency the online store will accept).